### ISSUE TITLE
Client freezes on sending message with malformed quoted code block

### ISSUE THREAD

**Issue by danpere** [mattermost/mattermost-server#8196]
#### Summary
Sending a message with an improperly quoted code block can take several seconds or freeze the client entirely without sending the message.

#### Steps to reproduce
Using Mattermost server v4.6.0 and client v4.0.0.

1. Send the following message:
> ````
> Test:
> > ```
> {
>   "Foo": true,
>   "Bar": {
>     "Key": "Baz"
>   },
>   "Foo": [
>     "Bar"
>   ],
>   "Baz": 2561,
>   "Bar": 5322,
>   "Foo": "Bar"
> }
> ```
> 
> Test
> ````

or this longer variant:

> ````
> Test:
> > ```
> {
>   "Foo": true,
>   "Bar": {
>     "Key": "Baz"
>   },
>   "Foo": [
>     "Bar"
>   ],
>   "Baz": 2561,
>   "Bar": 5322,
>   "Foo": "Bar"
> }
> {
>   "Foo": true,
>   "Bar": {
>     "Key": "Baz"
>   },
>   "Foo": [
>     "Bar"
>   ],
>   "Baz": 2561,
>   "Bar": 5322,
>   "Foo": "Bar"
> }
> ```
> 
> Test
> ````

2. Wait for message to send.

#### Expected behavior
I expect any reasonable length method to be sent quickly.

#### Observed behavior (that appears unintentional) 
The first message above takes ~12 seconds to send for me; I haven't waited long enough to know how long the second one would take, but in my experiments the time seems to approximately double with every two new lines in the code block. Changing the `"```"` line to `"> ```"` makes the message send instantly instead. Additionally, my attempts to simplify this case further (e.g. replacing the lines in the text block with just a single character each line with or without an indent), have caused the slowdown to appear to go away.

(Since Mattermost remembers the contents of the edit box, the freeze is not actually a huge problem: I could just refresh the client, edit the message, and try sending again. There's no loss of data here.)

#### Possible fixes
Sorry, I haven't looked at the code.

**Comment by ccbrown** [comment]
I can give you a simpler test case:

```
> ```
xxxxxxxxxxxx
xxxxxxxxxxxx
xxxxxxxxxxxx
xxxxxxxxxxxx
xxxxxxxxxxxx
xxxxxxxxxxxx
xxxxxxxxxxxx
xxxxxxxxxxxx
```

100% of the time you spend waiting is spent on a regular expression replacement. 😱

Fortunately, we shouldn't be using regular expressions for this to begin with, and this PR will fix eliminate them: https://github.com/mattermost/mattermost-webapp/pull/693

So this'll be fixed in 4.7. 😁

**Comment by hmhealey** [comment]
Thanks for the report @danpere. The PR to fix this has been merged, and it'll be available in our next release that's coming in a week or so
